Tools/cases_generator/generators_common.py

```diff
@@ -357,6 +359,19 @@ def reload_stack(
         self.emit_reload(storage)
         return True
 
+    def instruction_size(self,
+        tkn: Token,
+        tkn_iter: TokenIterator,
+        uop: Uop,
+        storage: Storage,
+        inst: Instruction | None
+    ) -> bool:
+        """Replace the INSTRUCTION_SIZE macro with the size of the current instruction."""
+        size = inst.size if inst else 1 + uop.size
```

Comment:
tier2 does not have access to the instruction here so I calculate the size as `1 + uop.size`. Does that make sense?

Reply:
Sorry this seems wrong, you need to emit the 1 + inline cache size not uop size

Reply:
Indeed, I see when comparing the output of `generated_cases.c.h` with `executor_cases.c.h` that the latter has the wrong instruction size. I fixed it (I think) by looking up the instructions for each `uop` in tier2 and passing it to the emitter.
Let me know if it's correct like that :)

Reply:
`INSTRUCTION_SIZE` is the size of the tier 1 instruction. This must be true even in tier 2.

> I fixed it (I think) by looking up the instructions for each uop in tier2 and passing it to the emitter.

That is correct, but it would be nice if we also checked that all matching tier 1 instructions are the same size.